### DISCUSSION TITLE
How to use ng deploy with CI systems like Github Actions

### DISCUSSION THREAD

**Discussion by OmarioHassan** [angular/angularfire#2957] [Q&A]
I'm trying to use ng deploy with token but I can't
Any help

**Comment by jamesdaniels** [comment]
Locally run `firebase login:ci` which will generate a CI token for you. Then take that token and set it as an environment variable (`FIREBASE_TOKEN`) in your CI system, if available use a "secret" environment token so it's encrypted.

**Reply by jamesdaniels** [reply]
For Github Actions here's the doc on encrypting environment variables https://docs.github.com/en/actions/reference/encrypted-secrets

**Reply by OmarioHassan** [reply]
@jamesdaniels 
![image](https://user-images.githubusercontent.com/8146344/133919160-125aff72-8664-4423-9a44-3c5292e0110a.png)
![image](https://user-images.githubusercontent.com/8146344/133919165-a43e8771-1852-4085-bd2c-d54c7aac5a36.png)
![image](https://user-images.githubusercontent.com/8146344/133919175-f0e6abe2-4e85-4e33-8cd6-e7ca6960b593.png)

**Reply by jamesdaniels** [reply]
It needs to be an environment variable, not passed as an option.

**Reply by OmarioHassan** [reply]
So how to connect GitHub secret variable with local environment variable
Is there a code example

**Reply by jamesdaniels** [reply]
> So how to connect GitHub secret variable with local environment variable
> 
> Is there a code example

look at the github actions docs, it's pretty explicit on how to use env variables

**Reply by JasonGoemaat** [reply]
> So how to connect GitHub secret variable with local environment variable Is there a code example

This is what I use:

```yml
  - name: Firebase Deploy
    run: firebase deploy --token ${{ secrets.FIREBASE_TOKEN }}
````

**Comment by OmarioHassan** [comment]
@jamesdaniels after fixing the token issue, I faced a new one 

![image](https://user-images.githubusercontent.com/8146344/134229783-d1a228c0-4e3a-4d2a-8f4e-b3c010eccb3e.png)

although it works well when try ng deploy locally

![image](https://user-images.githubusercontent.com/8146344/134229918-357c6a4d-05a2-4379-a215-16bfbd684da8.png)

**Reply by jamesdaniels** [reply]
You need to set that `public` directory in the hosting config to whatever the directory your browser bundle is built / been moved to. Use `ng deploy --preview` locally to see where those artifacts are being placed.

**Reply by OmarioHassan** [reply]
Here are logs and no errors for `ng deploy --preview`

![image](https://user-images.githubusercontent.com/8146344/134389140-a13d2089-9a67-4cde-8ac1-f327f3905054.png)

and this is for the browser logs 

![image](https://user-images.githubusercontent.com/8146344/134389221-f0621bb0-aea1-467b-b8c7-8f8366c7e8ce.png)

**Reply by OmarioHassan** [reply]
dist output looks well 

![image](https://user-images.githubusercontent.com/8146344/134389455-e4016603-c602-4b83-a573-f95b9c5cf72e.png)

**Reply by OmarioHassan** [reply]
@jamesdaniels any help 😊

**Comment by OmarioHassan** [comment]
@jamesdaniels Any Support?
